### ISSUE TITLE
Code generation fails after upgrade from 5.1.1 to 5.1.2 (vertx-jooq-classic-reactive)

### ISSUE THREAD

**Issue by Dudeplayz** [jklingsporn/vertx-jooq#159]
After I upgraded from `5.1.1` to `5.1.2` the code generation fails. If I change it back, everything works again. The StackTrace is appended at the end.

Versions:
```
vertx-jooq-classic-reactive : 5.1.2
jooq : 3.13.2
vertx : 3.9.1
postgres : 42.2.14
```

<details>
  <summary>StackTrace</summary>

```
Failed to execute goal org.jooq:jooq-codegen-maven:3.13.2:generate (default) on project bita-backend: Error running jOOQ code generation tool: Error generating code for catalog: Error generating code for schema bita: character to be escaped is missing -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.jooq:jooq-codegen-maven:3.13.2:generate (default) on project bita-backend: Error running jOOQ code generation tool
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: org.apache.maven.plugin.MojoExecutionException: Error running jOOQ code generation tool
    at org.jooq.codegen.maven.Plugin.execute (Plugin.java:210)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: org.jooq.codegen.GeneratorException: Error generating code for catalog 
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:399)
    at org.jooq.codegen.GenerationTool.run0 (GenerationTool.java:845)
    at org.jooq.codegen.GenerationTool.run (GenerationTool.java:222)
    at org.jooq.codegen.GenerationTool.generate (GenerationTool.java:217)
    at org.jooq.codegen.maven.Plugin.execute (Plugin.java:207)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: org.jooq.codegen.GeneratorException: Error generating code for schema bita
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:478)
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:394)
    at org.jooq.codegen.GenerationTool.run0 (GenerationTool.java:845)
    at org.jooq.codegen.GenerationTool.run (GenerationTool.java:222)
    at org.jooq.codegen.GenerationTool.generate (GenerationTool.java:217)
    at org.jooq.codegen.maven.Plugin.execute (Plugin.java:207)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: java.lang.IllegalArgumentException: character to be escaped is missing
    at java.util.regex.Matcher.appendExpandedReplacement (Matcher.java:1020)
    at java.util.regex.Matcher.appendReplacement (Matcher.java:998)
    at java.util.regex.Matcher.replaceAll (Matcher.java:1182)
    at java.lang.String.replaceAll (String.java:2126)
    at io.github.jklingsporn.vertx.jooq.generate.builder.ComponentBasedVertxGenerator.generateTargetFile (ComponentBasedVertxGenerator.java:232)
    at io.github.jklingsporn.vertx.jooq.generate.builder.VertxGeneratorBuilder$ExecutionStepImpl.lambda$withPostgresReactiveDriver$22 (VertxGeneratorBuilder.java:291)
    at io.github.jklingsporn.vertx.jooq.generate.builder.DelegatingVertxGenerator.lambda$writeExtraData$0 (DelegatingVertxGenerator.java:99)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:195)
    at java.util.ArrayList$ArrayListSpliterator.forEachRemaining (ArrayList.java:1654)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:484)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:474)
    at java.util.stream.ReduceOps$ReduceOp.evaluateSequential (ReduceOps.java:913)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.collect (ReferencePipeline.java:578)
    at io.github.jklingsporn.vertx.jooq.generate.builder.DelegatingVertxGenerator.writeExtraData (DelegatingVertxGenerator.java:99)
    at io.github.jklingsporn.vertx.jooq.generate.VertxGenerator.writeExtraData (VertxGenerator.java:242)
    at io.github.jklingsporn.vertx.jooq.generate.VertxGenerator.generateDaos (VertxGenerator.java:238)
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:531)
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:473)
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:394)
    at org.jooq.codegen.GenerationTool.run0 (GenerationTool.java:845)
    at org.jooq.codegen.GenerationTool.run (GenerationTool.java:222)
    at org.jooq.codegen.GenerationTool.generate (GenerationTool.java:217)
    at org.jooq.codegen.maven.Plugin.execute (Plugin.java:207)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
```
</details>

**Comment by jklingsporn** [comment]
This is a regression from https://github.com/jklingsporn/vertx-jooq/issues/151. Can you share some further information about your configuration? Especially how your package name for the generated classes should look like.

**Comment by Dudeplayz** [comment]
Hey, sorry for the late reply, and thanks for the fast response.
I think you mean the maven configuration:

<details>
<summary>Used maven config (groupId blackened)</summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <artifactId>bita-backend</artifactId>
    <version>${backend.version}</version>
    <packaging>jar</packaging>

    <name>Bita-Backend</name>

    <parent>
        <groupId>...</groupId>
        <artifactId>bita-parent</artifactId>
        <version>0.0.1-SNAPSHOT</version>
    </parent>

    <properties>
        <!--    jOOQ    -->
        <postgres.version>42.2.14</postgres.version>
        <!--    DEPENDENCIES    -->
        <jooq.version>3.13.2</jooq.version>
        <vertx-jooq.version>5.1.1</vertx-jooq.version>
    </properties>

    <dependencies>
        <dependency>
            <groupId>...</groupId>
            <artifactId>bita-core</artifactId>
            <version>${core.version}</version>
        </dependency>
        <dependency>
            <groupId>io.vertx</groupId>
            <artifactId>vertx-pg-client</artifactId>
        </dependency>
        <!--    jOOQ    -->
        <dependency>
            <groupId>org.jooq</groupId>
            <artifactId>jooq</artifactId>
            <version>${jooq.version}</version>
        </dependency>
        <dependency>
            <groupId>io.github.jklingsporn</groupId>
            <artifactId>vertx-jooq-classic-reactive</artifactId>
            <version>${vertx-jooq.version}</version>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>build-helper-maven-plugin</artifactId>
                <version>3.1.0</version>
                <executions>
                    <execution>
                        <phase>generate-sources</phase>
                        <goals>
                            <goal>add-source</goal>
                        </goals>
                        <configuration>
                            <sources>
                                <source>src/jooq/java</source>
                            </sources>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>

    <profiles>
        <profile>
            <id>jooq-generate</id>
            <properties>
                <!--    jOOQ properties   -->
                <jooq.driver>org.postgresql.Driver</jooq.driver>
                <jooq-generated.dir>src/jooq/java</jooq-generated.dir>
                <jooq-generated.package>...backend.db</jooq-generated.package>
                <!--    Placeholder, uncomment for external pom.properties file    -->
<!--                <postgres.url></postgres.url>-->
<!--                <postgres.user></postgres.user>-->
<!--                <postgres.password></postgres.password>-->
            </properties>
            <build>
                <plugins>
                    <plugin>
                        <groupId>org.codehaus.mojo</groupId>
                        <artifactId>properties-maven-plugin</artifactId>
                        <version>1.0.0</version>
                        <executions>
                            <execution>
                                <phase>initialize</phase>
                                <goals>
                                    <goal>read-project-properties</goal>
                                </goals>
                                <configuration>
                                    <files>
                                        <file>pom.properties</file>
                                    </files>
                                </configuration>
                            </execution>
                        </executions>
                    </plugin>
                    <plugin>
                        <!-- Specify the maven code generator plugin -->
                        <groupId>org.jooq</groupId>
                        <artifactId>jooq-codegen-maven</artifactId>
                        <version>${jooq.version}</version>

                        <!-- The plugin should hook into the generate goal -->
                        <executions>
                            <execution>
                                <goals>
                                    <goal>generate</goal>
                                </goals>
                            </execution>
                        </executions>

                        <dependencies>
                            <dependency>
                                <groupId>org.postgresql</groupId>
                                <artifactId>postgresql</artifactId>
                                <version>${postgres.version}</version>
                            </dependency>
                            <dependency>
                                <groupId>io.github.jklingsporn</groupId>
                                <artifactId>vertx-jooq-generate</artifactId>
                                <version>${vertx-jooq.version}</version>
                            </dependency>
                        </dependencies>

                        <!-- Specify the plugin configuration.
                             The configuration format is the same as for the standalone code generator -->
                        <configuration>
                            <!-- JDBC connection parameters -->
                            <jdbc>
                                <driver>${jooq.driver}</driver>
                                <url>${postgres.url}</url>
                                <user>${postgres.user}</user>
                                <password>${postgres.password}</password>
                            </jdbc>

                            <!-- Generator parameters -->
                            <generator>
                                <name>io.github.jklingsporn.vertx.jooq.generate.classic.ClassicReactiveVertxGenerator
                                </name>
                                <!-- use 'io.github.jklingsporn.vertx.jooq.generate.classic.ClassicReactiveGuiceVertxGenerator' to enable Guice DI -->
                                <database>
                                    <name>org.jooq.meta.postgres.PostgresDatabase</name>
                                    <includes>.*</includes>
                                    <schemata>
                                        <schema>
                                            <inputSchema>bita</inputSchema>
                                            <outputSchema>bita</outputSchema>
                                        </schema>
                                        <schema>
                                            <inputSchema>security</inputSchema>
                                            <outputSchema>security</outputSchema>
                                        </schema>
                                    </schemata>
                                    <unsignedTypes>false</unsignedTypes>
                                </database>
                                <target>
                                    <packageName>${jooq-generated.package}</packageName>
                                    <directory>${jooq-generated.dir}</directory>
                                </target>
                                <generate>
                                    <interfaces>true</interfaces>
                                    <daos>true</daos>
                                    <fluentSetters>true</fluentSetters>
                                </generate>

                                <strategy>
                                    <name>io.github.jklingsporn.vertx.jooq.generate.VertxGeneratorStrategy</name>
                                </strategy>
                            </generator>
                        </configuration>
                    </plugin>
                </plugins>
            </build>
        </profile>
    </profiles>

</project>

```

</details>

**Comment by jklingsporn** [comment]
To me `<jooq-generated.package>...backend.db</jooq-generated.package>` looks weird.

**Comment by Dudeplayz** [comment]
I blackened the groupId, its something like `<jooq-generated.package>a.b.c.backend.db</jooq-generated.package>`

**Comment by ernieKovak** [comment]
We see the same thing. Here's how we set the pkg name:
`
<target>
    <packageName>com.company.project.domain2.jooq</packageName>
    <directory>${basedir}/src/main/java</directory>
</target>
`

**Comment by yufield** [comment]
https://github.com/jklingsporn/vertx-jooq/blob/6364c2304aeb0b365e054aacb21e7bb895df4905/vertx-jooq-generate/src/main/java/io/github/jklingsporn/vertx/jooq/generate/builder/ComponentBasedVertxGenerator.java#L237
In my case,it works after i changed
`File.separator` to`Matcher.quoteReplacement(File.separator)`

**Comment by Dudeplayz** [comment]
@yufield how have you fixed this?
I need to use 5.2.0, but can't because the RowMapper generation fails...

**Comment by jklingsporn** [comment]
Should be fixed in 5.2.1. However I cannot test this as it appears to be a windows problem. Can anyone confirm it is fixed?

**Comment by Dudeplayz** [comment]
I have done a short check and it seems to work. Thank you 👍.
